src/components/ProviderSettings/connect.js

```diff
@@ -0,0 +1,23 @@
+import { connect } from 'react-redux';
+import getProviderModel, * as providerActionCreators from '../../store/models/provider';
+
+export function mapStateToProps(state, props) {
+  const provider = getProviderModel(state);
+
+  return {
+    loading: provider.isLoading(),
+    error: provider.getError(),
+    currentProvider: provider.getCurrentProvider(),
+  };
+}
+
+export function mapDispatchToProps(dispatch) {
+  return {
+    setProvider: options => dispatch(providerActionCreators.setProvider(options)),
+  };
+}
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
```

Comment:
As you are not wrapping any `props` to the action creators at the connect level, you can consider doing `connect(mapStateToProps, providerActionCreators)` directly.

Reply:
I am not sure I remember exactly why i didn't do it directly, but maybe it had to do with the testing. In that case , only mapStateToProps needs to be tested right ?